modules/rosa-sts-creating-a-cluster-quickly.adoc

```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// * rosa_getting_started_sts/rosa_creating_a_cluster_with_sts/rosa-sts-creating-a-cluster-quickly.adoc
+
+[id="rosa-sts-creating-cluster-using-defaults{context}"]
+= Creating a ROSA cluster with STS using the default options
+
+Through the {product-title} CLI (`rosa`), you can quickly create an OpenShift cluster that uses the AWS Security Token Service (STS). The `--sts` argument enables you to deploy a cluster with STS using the default installation options.
+
+Additionally, you can use `auto` mode to immediately create the required AWS Identity and Access Management (IAM) resources using the current AWS account.
+
+[IMPORTANT]
+====
+Only public and AWS PrivateLink clusters are supported with STS. Regular private clusters (non-PrivateLink) are not available for use with STS.
+====
+
+.Prerequisites
+
+* You have completed the AWS prerequisites for ROSA with STS.
+* You have available AWS service quotas.
+* You have enabled the ROSA service in the AWS Console.
+* You have installed and configured the latest AWS, ROSA, and `oc` CLIs on your installation host.
+
+[NOTE]
+====
+link:https://docs.aws.amazon.com/vpc/latest/userguide/vpc-sharing.html[AWS Shared VPCs] are not currently supported for ROSA installations.
+====
+
+.Procedure
+
+. Create the required account-wide roles and policies, including the Operator policies:
++
+[source,terminal]
+----
+$ rosa create account-roles --mode auto --version 4.8 <1>
+----
+<1> You must specify an OpenShift release by using the `--version` option. The version that is installed when a ROSA cluster is created is determined by the roles and policies that are used.
++
+[NOTE]
+====
+When using `auto` mode, you can optionally specify the `-y` argument to bypass the interactive prompts and automatically confirm operations.
+====
+
+. Create a cluster with STS using the defaults:
++
+[source,terminal]
+----
+$ rosa create cluster --cluster-name <cluster_name> --sts <1>
+----
+<1> Replace `<cluster_name>` with the name of your cluster.
++
+[IMPORTANT]
+====
+You must complete the following steps to create the Operator IAM roles and the OpenID Connect (OIDC) provider to move the state of the cluster from `pending` to `ready`.
+====
+
+. Create the cluster-specific Operator IAM roles:
++
+[source,terminal]
+----
+$ rosa create operator-roles --mode auto --cluster <cluster_name|cluster_id>
+----
+
+. Create the OpenID Connect (OIDC) provider that the cluster Operators will use to authenticate:
++
+[source,terminal]
+----
+$ rosa create oidc-provider --mode auto --cluster <cluster_name|cluster_id>
+----
+
+. Check the status of your cluster. The `State` field changes from `pending` to `installing` to `ready`:
```

Comment:
There is a new state _before_ the user creates the OIDC provider: `waiting`. Might be helpful to call it out here.

Reply:
@vkareh thank you for highlighting this. I just tested a deployment with the `rosa` 1.1.3 client, to see what order the status updates appear in. In my test, the state changed as follows:

* `pending (Waiting for OIDC configuration)`
* `pending (Preparing account)`
* `installing (DNS setup in progress)`
* `installing*
* `ready`

The query loop I created had a 5 second interval, so I'm not sure if I missed any states that changed within a single command interval.

Is this being changed further or does that reflect the current sequence of states in the output?

Given that there are 5 items in that list, we could remove the state reference from this step and keep it simple. Then, we could highlight the state changes in the note about states that is in step 5.

Reply:
Strange, the first step should read
- `waiting (Waiting for OIDC configuration)`
unless you have a custom build or are using a different OCM environment - this change went in with v1.1.3

Maybe remove it for now until this is sorted out. Since those steps can change based on changes in OCM, maybe it's simpler to not spell them out